### PR TITLE
build: fix size-limit check and add GitHub action

### PR DESCRIPTION
**Pull request by denkristoffer**
# Purpose of PR

The size-limit check is broken in `main`, this branch fixes that and adds a GitHub action that should comment with size differences on pull requests 🙂 

FYI this PR will not go green until after it's been merged due to the way the action works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -149,17 +151,18 @@
...
     "semver": "^7.3.5",
+    "size-limit": "^8.0.1",
     "style-loader": "^2.0.0",
     "turbo": "^1.0.6",
     "typescript": "4.7.3"
   },
   "engines": {
     "node": ">=14.17.0",
-    "yarn": ">=1.19.0"
+    "yarn": ">=1.18.0"
```
Comment (mshaaban0) [inline package.json:161]: downgrade, more compatibility 👍🏼

File: packages/forma-36-react-components/package.json [hunk 2]
```diff
@@ -13,8 +13,7 @@
   "browserslist": "extends @contentful/browserslist-config",
   "scripts": {
     "build:clean": "rm -rf dist",
-    "build": "yarn build:clean && microbundle src/index.ts && yarn size",
-    "size": "size-limit",
+    "build": "yarn build:clean && microbundle src/index.ts",
```
Comment (mshaaban0) [inline packages/forma-36-react-components/package.json:16]: Q: why are we using `microbundle` (Rollup) only for this package and not parcel like others?

File: .github/workflows/size-limit.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+name: 'size'
+
+on: pull_request
+  branches:
+    - main
```
Comment (mshaaban0) [inline .github/workflows/size-limit.yml:5]: Heads up there is an error on this part, the action is not reporting back to the PR
https://github.com/contentful/forma-36/actions/runs/2962500861/workflow

File: packages/forma-36-react-components/package.json (hunk 2 above)
Comment (denkristoffer) [reply packages/forma-36-react-components/package.json:16]: I've no idea, I tried to track it down and it comes from the original v4 pull request. I wanted to look into it in a separate PR 🤷

File: package.json (hunk 1 above)
Comment (denkristoffer) [reply package.json:161]: I'm not sure why we required 1.19 in the first place, any idea?

**Review by bgutsol — APPROVED** [review]

File: .github/workflows/size-limit.yml (hunk 3 above)
Comment (mshaaban0) [reply .github/workflows/size-limit.yml]: @denkristoffer it's still failing to build 🤔

Comment (denkristoffer) [reply .github/workflows/size-limit.yml]: @mshaaban0 It's not going to work for this PR, but you can see it working here: https://github.com/contentful/forma-36/runs/8136279821?check_suite_focus=true

The reason it fails here is that the action works by running `build` and then `npx size-limit --json` first on the branch of the PR, then on `main`, and it compares the two size results. Before this PR was merged the configuration for `size-limit` was broken on the `main` branch, so it couldn't compare the results and then exited with an error. It should work for future PRs, although it looks like the config needs to be tweaked.

**Pull request merged**